test/e2e/resize_nodes.go

```
@@ -0,0 +1,331 @@
+	}
+	return fmt.Errorf("Timeout waiting for node instance group size to be %v", size)
+}
+
+func waitForClusterSize(c *client.Client, size int) error {
+	for start := time.Now(); time.Since(start) < 2*time.Minute; time.Sleep(20 * time.Second) {
+		nodes, err := c.Nodes().List(labels.Everything(), fields.Everything())
+		if err != nil {
+			Logf("Failed to list nodes: %v", err)
+			continue
+		}
+		if len(nodes.Items) == size {
+			Logf("Cluster has reached the desired size %v", size)
+			return nil
+		}
+		Logf("Waiting for cluster size %v, current size %v", size, len(nodes.Items))
+	}
+	return fmt.Errorf("Timeout waiting for cluster size to be %v", size)
+}
+
+func newService(name string) *api.Service {
+	return &api.Service{
+		ObjectMeta: api.ObjectMeta{
+			Name: "test-service",
+		},
+		Spec: api.ServiceSpec{
+			Selector: map[string]string{
+				"name": name,
+			},
+			Ports: []api.ServicePort{{
+				Port:       9376,
+				TargetPort: util.NewIntOrStringFromInt(9376),
+			}},
+		},
+	}
+}
+
+func createService(c *client.Client, ns, name string) {
+	_, err := c.Services(ns).Create(newService(name))
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func newReplicationController(name string, replicas int, image string) *api.ReplicationController {
+	return &api.ReplicationController{
+		ObjectMeta: api.ObjectMeta{
+			Name: name,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: replicas,
+			Selector: map[string]string{
+				"name": name,
+			},
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: map[string]string{"name": name},
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  name,
+							Image: image,
+							Ports: []api.ContainerPort{{ContainerPort: 9376}},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createServeHostnameReplicationController(c *client.Client, ns, name string, replicas int) *api.ReplicationController {
```

Please return an error as well rather than checking it here.